packages/tools/test/EraserTool_test.js

```diff
@@ -0,0 +1,208 @@
+  imageLoader,
+  metaData,
+  volumeLoader,
+} = cornerstone3D;
+
+const { Events, ViewportType } = Enums;
+
+const {
+  LengthTool,
+  ToolGroupManager,
+  Enums: csToolsEnums,
+  annotation,
+} = csTools3d;
+
+const { Events: csToolsEvents } = csToolsEnums;
+
+const {
+  fakeImageLoader,
+  fakeVolumeLoader,
+  fakeMetaDataProvider,
+  createNormalizedMouseEvent,
+} = testUtils;
+
+const renderingEngineId = utilities.uuidv4();
+
+const viewportId = 'VIEWPORT';
+
+function createViewport(renderingEngine, viewportType, width, height) {
+  const element = document.createElement('div');
+
+  element.style.width = `${width}px`;
+  element.style.height = `${height}px`;
+  document.body.appendChild(element);
+
+  renderingEngine.setViewports([
+    {
+      viewportId: viewportId,
+      type: viewportType,
+      element,
+      defaultOptions: {
+        background: [1, 0, 1], // pinkish background
+        orientation: Enums.OrientationAxis.AXIAL,
+      },
+    },
+  ]);
+  return element;
+}
+
+const volumeId = `fakeVolumeLoader:volumeURI_100_100_10_1_1_1_0`;
+
+describe('EraserTool:', () => {
+  beforeAll(() => {
+    cornerstone3D.setUseCPURendering(false);
+  });
+
+  describe('Cornerstone Tools: -- Eraser', () => {
+    beforeEach(function () {
+      csTools3d.init();
+      csTools3d.addTool(EraserTool);
+      csTools3d.addTool(LengthTool);
+
+      cache.purgeCache();
+      this.DOMElements = [];
+
+      this.stackToolGroup = ToolGroupManager.createToolGroup('stack');
+      this.stackToolGroup.addTool(EraserTool.toolName, {});
+      this.stackToolGroup.setToolActive(EraserTool.toolName, {
+        bindings: [{ mouseButton: 1 }],
+      });
+
+      this.renderingEngine = new RenderingEngine(renderingEngineId);
+      imageLoader.registerImageLoader('fakeImageLoader', fakeImageLoader);
+      volumeLoader.registerVolumeLoader('fakeVolumeLoader', fakeVolumeLoader);
+      metaData.addProvider(fakeMetaDataProvider, 10000);
+    });
+
+    afterEach(function () {
+      csTools3d.destroy();
+      eventTarget.reset();
+      cache.purgeCache();
+
+      this.renderingEngine.destroy();
+      metaData.removeProvider(fakeMetaDataProvider);
+      imageLoader.unregisterAllImageLoaders();
+      ToolGroupManager.destroyToolGroup('stack');
+
+      this.DOMElements.forEach((el) => {
+        if (el.parentNode) {
+          el.parentNode.removeChild(el);
+        }
+      });
+    });
+
+    it('Should successfully delete a length annotation on a canvas with mouse down - 512 x 128', function (done) {
+      const element = createViewport(
+        this.renderingEngine,
+        ViewportType.STACK,
+        512,
+        128
+      );
+
+      this.DOMElements.push(element);
+
+      const imageId1 = 'fakeImageLoader:imageURI_64_64_10_5_1_1_0';
+      const vp = this.renderingEngine.getViewport(viewportId);
+
+      const addEventListenerForAnnotationRemoved = () => {
+        element.addEventListener(csToolsEvents.ANNOTATION_REMOVED, () => {
```

Comment:
I think here instead of adding the listener to `element`, `eventTarget` should be used.